rmf_task/src/rmf_task/agv/TaskPlanner.cpp

```diff
@@ -123,6 +120,22 @@ auto TaskPlanner::Configuration::planner(
   return *this;
 }
 
+//==============================================================================
+std::shared_ptr<rmf_task::CostCalculator>
+TaskPlanner::Configuration::cost_calculator() const
+{
+  return _pimpl->cost_calculator;
+}
+
+//==============================================================================
+auto TaskPlanner::Configuration::cost_calculator(
+  std::shared_ptr<rmf_task::CostCalculator> cost_calculator) -> Configuration&
+{
+  if (cost_calculator)
+    _pimpl->cost_calculator = cost_calculator;
```

Comment:
I might suggest allowing this to have a `nullptr` value and then have the planner automatically select a default if the user provided a `nulltpr`.

In cases where `nullptr` inputs are unacceptable, I think it's better to throw an exception rather than silently do nothing.

Reply:
Done https://github.com/osrf/rmf_core/pull/278/commits/18beb68180600b855b48acdb86176e2ed85d6855